drivers/gnss/gnss_nmea_generic.c

```diff
@@ -148,24 +154,49 @@ static int gnss_nmea_generic_init(const struct device *dev)
 		return ret;
 	}
 
+#if CONFIG_PM_DEVICE
+	pm_device_init_suspended(dev);
+#else
 	ret = gnss_nmea_generic_resume(dev);
 	if (ret < 0) {
 		return ret;
 	}
+#endif
 
 	return 0;
 }
 
+#if CONFIG_PM_DEVICE
+static int gnss_nmea_generic_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		return gnss_nmea_generic_resume(dev);
+	default:
+		return -ENOTSUP;
+	}
```

Comment:
not too convinced about this one, late initialization can already be done with https://github.com/zephyrproject-rtos/zephyr/pull/67335, this seems like a misuse of the PM framework, I'd imagine PM for this drvier to use either a custom modem script for suspend/resume or a gpio (depending on what the device uses)

Reply:
Very good point, but I still think this is a correct implementation. Or at least a start at getting PM implemented for the driver. Maybe more states need to be implemented here for a start?

Reply:
sure, if we are fine with this as a start though I think that this should have a boolean to check that `gnss_nmea_generic_resume` only gets called once then, just making sure it does not get reinitialized multiple times if the device get few cycles of resume/suspend

Reply:
I was hoping we would create specific drivers for GNSS modems to handle PM, as many GNSS modems use chat commands to resume/suspend alongside pins (and some of them behave really oddly, requiring delays here an there to not deadlock...).

Reply:
> late initialization can already be done with https://github.com/zephyrproject-rtos/zephyr/pull/67335

I wasn't aware of this. Good to know.

Though as @rerickson1 pointed out, this is a (admittedly rudimentary) `start at getting PM implemented for the driver`. My point with this PR is to flush out some changes I have locally that are related to work I have started but am not able to continue at the moment.

> I'd imagine PM for this drvier to use either a custom modem script for suspend/resume or a gpio (depending on what the device uses)

The changes here do in fact introduce a custom resume script. It's just that currently there is only the `gnss-nmea-generic` devicetree node, and for it I made the script be empty (to not change the initialization behavior and I wouldn't know what to put anyway).

> though I think that this should have a boolean to check that gnss_nmea_generic_resume only gets called once then, just making sure it does not get reinitialized multiple times if the device get few cycles of resume/suspend

Actually `pm_device` takes care of that already. In `pm_device_action_run()`, before calling the PM action callback of the driver, it validates that the current state is not already the one that will be set. And the current state is only changed if the action callback returns no error. As the driver returns `-ENOTSUP` for PM actions other than `PM_DEVICE_ACTION_RESUME`, the state won't ever be changed to something else than `PM_DEVICE_STATE_ACTIVE`, and only one `PM_DEVICE_ACTION_RESUME` can be received by the callback.

> I was hoping we would create specific drivers for GNSS modems to handle PM, as many GNSS modems use chat commands to resume/suspend alongside pins (and some of them behave really oddly, requiring delays here an there to not deadlock...).

I saw that PM had been directly implemented in `GNSS_QUECTEL_LCX6G`, so I did the same here. Anyway I imagine this doesn't preclude changing how this is done at a later point.